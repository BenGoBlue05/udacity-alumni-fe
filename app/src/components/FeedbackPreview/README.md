## FeedbackPreview Component
A component that ...

### Example

```js
<FeedbackPreview />
```

### Props

| Prop          | Type     | Default     | Possible Values
| ------------- | -------- | ----------- | ---------------------------------------------
| **myProp**    | String   |             | Any string value


### Other Information
