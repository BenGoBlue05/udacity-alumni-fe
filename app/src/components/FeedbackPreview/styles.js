import styled from 'styled-components';
import Box from 'grommet-udacity/components/Box';

export const BoxWrapper = styled(Box)`
  display: flex;
  align-items: center;
  justify-content: space-between;
  flex-direction: row !important;
  width: 100%;
`;
