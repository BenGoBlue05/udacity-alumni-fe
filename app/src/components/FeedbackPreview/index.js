import React, { PropTypes } from 'react';
import Layer from 'grommet-udacity/components/Layer';
import Heading from 'grommet-udacity/components/Heading';
import Box from 'grommet-udacity/components/Box';
import Label from 'grommet-udacity/components/Label';
import Timestamp from 'grommet-udacity/components/Timestamp';
import Anchor from 'grommet-udacity/components/Anchor';
import { BoxWrapper } from './styles';

const FeedbackPreview = ({
  isVisible,
  onClose,
  feedbackItem,
}) => (
  <Layer
    onClose={onClose}
    closer
    align="center"
    hidden={!isVisible}
  >
    {feedbackItem &&
      <Box
        size="large"
        pad="large"
      >
        <Heading align="center">
          {feedbackItem.user.name}
        </Heading>
        <BoxWrapper>
          <Label>
            Email:
          </Label>
          <Heading align="center" tag="h4">
            <Anchor href={`mailto:${feedbackItem.user.email}`}>
              {feedbackItem.user.email}
            </Anchor>
          </Heading>
        </BoxWrapper>
        <BoxWrapper>
          <Label>
            Date:
          </Label>
          <Timestamp value={feedbackItem.created_at} fields={['date', 'month', 'year']} />
        </BoxWrapper>
        <BoxWrapper>
          <Label style={{ flex: 1 }}>
            Description:
          </Label>
          <Heading align="center" tag="h4">
            {feedbackItem.description}
          </Heading>
        </BoxWrapper>
        <BoxWrapper>
          <Label style={{ flex: 1 }}>
            Url:
          </Label>
          <Heading align="center" tag="h4">
            {feedbackItem.url}
          </Heading>
        </BoxWrapper>
      </Box>
    }
  </Layer>
);

FeedbackPreview.propTypes = {
  isVisible: PropTypes.bool.isRequired,
  onClose: PropTypes.func.isRequired,
  feedbackItem: PropTypes.object.isRequired,
};

export default FeedbackPreview;
