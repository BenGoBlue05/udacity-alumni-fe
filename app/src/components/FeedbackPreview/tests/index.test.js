import FeedbackPreview from '../index';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';

describe('<FeedbackPreview />', () => {
  it('should render with default props', () => {
    const wrapper = shallow(
      <FeedbackPreview />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
