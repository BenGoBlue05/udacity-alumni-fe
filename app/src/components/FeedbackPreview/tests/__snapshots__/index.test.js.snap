exports[`<FeedbackPreview /> should render with default props 1`] = `
<Layer
  align="center"
  closer={true}
  hidden={true} />
`;
